src/components/AttachmentCarousel/AttachmentCarouselView/index.js

```diff
@@ -0,0 +1,152 @@
+import React, {useRef, useCallback} from 'react';
+import {View, FlatList, PixelRatio} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../../styles/styles';
+import CarouselActions from '../CarouselActions';
+import AttachmentView from '../../AttachmentView';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../withWindowDimensions';
+import CarouselButtons from '../CarouselButtons';
+
+const propTypes = {
+    // eslint-disable-next-line react/forbid-prop-types
+    carouselState: PropTypes.object.isRequired,
+
+    updatePage: PropTypes.func.isRequired,
+    toggleArrowsVisibility: PropTypes.func.isRequired,
+    autoHideArrow: PropTypes.func.isRequired,
+    cancelAutoHideArrow: PropTypes.func.isRequired,
+
+    ...windowDimensionsPropTypes,
+};
+
+function AttachmentCarouselView(props) {
+    const scrollRef = useRef(null);
+    const viewabilityConfig = {
+        // To facilitate paging through the attachments, we want to consider an item "viewable" when it is
+        // more than 90% visible. When that happens we update the page index in the state.
+        itemVisiblePercentThreshold: 95,
+    };
```

Comment:
I still don't get what 90% visible means? Does that mean 90% of the image is currently being rendered (so the user is zoomed in slightly?)

Reply:
As described [here](https://reactnative.dev/docs/flatlist#viewabilityconfig), i suppose it means that as long as less than 90% of a new page (new attachment image) is visible on screen, we don't switch the page. And if more of the new page is shown, we switch

Reply:
We haven't changed the numbers here, that's all from [code on `main`](https://github.com/Expensify/App/blob/158c34bc07477edcd74db91f000254e2261d5538/src/components/AttachmentCarousel/index.js#L53-L57)